pilot/pkg/model/service.go

```diff
@@ -1441,7 +1440,14 @@ func (s *Service) Equals(other *Service) bool {
 
 // DeepCopy creates a clone of IstioEndpoint.
 func (ep *IstioEndpoint) DeepCopy() *IstioEndpoint {
-	return copyInternal(ep).(*IstioEndpoint)
+	out := *ep
+	out.Labels = maps.Clone(ep.Labels)
+	out.Addresses = slices.Clone(ep.Addresses)
```

Comment:
seems this is not need, we donot mutate it later after created.

Reply:
I think it might be better to keep the DeepCopy semantics here.